packages/bun-usockets/src/loop.c

```diff
@@ -391,7 +391,40 @@ void us_internal_dispatch_ready_poll(struct us_poll_t *p, int error, int eof, in
                       const int recv_flags = MSG_DONTWAIT | MSG_NOSIGNAL;
                     #endif
 
-                    int length = bsd_recv(us_poll_fd(&s->p), loop->data.recv_buf + LIBUS_RECV_BUFFER_PADDING, LIBUS_RECV_BUFFER_LENGTH, recv_flags);
+                    int length;
+                    #if !defined(_WIN32)
+                    if(s->context->is_ipc) {
```

Comment:
we still have 4 more bits in the socket flags struct. can we avoid the pointer indirection?
```

struct us_socket_flags {
    /* If true, the readable side is paused */
    bool is_paused: 1;
    /* Allow to stay alive after FIN/EOF */
    bool allow_half_open: 1; 
    /* 0 = not in low-prio queue, 1 = is in low-prio queue, 2 = was in low-prio queue in this iteration */
    unsigned char low_prio_state: 2; 

} __attribute__((packed));
```